trpcultivate_phenotypes/src/Plugin/Validators/ValidHeaders.php

```diff
@@ -51,75 +40,76 @@ public static function create(ContainerInterface $container, array $configuratio
 
   /**
    * Validate the header row.
+   *
    * Checks include:
-   *  - Each header value is a non-empty string.
-   *  - No header is missing.
-   *  - The order of headers defined by the Importer should match the order of headers array.
+   * - Each header value is not empty.
+   * - No expected header is missing.
+   * - The order of headers defined by the Importer should match the input.
    *
    * @param array $headers
-   *   An array of headers created by splitting the first line of the data file into separate values.
-   *   The index of the header represents the order they appear in the line.
+   *   An array created by splitting the first line of the data file into
+   *   values. The index of each value represents the order it appears in.
    *
    * @return array
    *   An associative array with the following keys.
-   *     - case: a developer focused string describing the case checked.
-   *     - valid: either TRUE or FALSE depending on if the header is valid or not.
-   *     - failedItems: the failed headers. This will be an empty array if the header was valid.
+   *   - 'case': a developer-focused string describing the case checked.
+   *   - 'valid': TRUE if the header is valid, FALSE otherwise.
+   *   - 'failedItems': an array of items that failed with any of the following
+   *      keys. This is an empty array if the header passed validation.
+   *      - 'headers': A string indicating the header row is empty.
+   *      - the headers input array.
    */
   public function validateRow($headers) {
     $input_headers = $headers;
 
-    // Parameter check, verify that the headers array input is not an empty array.
+    // Parameter check, verify that the headers array input is not empty.
     if (empty($headers)) {
       // Headers array is an empty array.
       return [
         'case' => 'Header row is an empty value',
         'valid' => FALSE,
-        'failedItems' => ['headers' => 'headers array is an empty array']
+        'failedItems' => [
+          'headers' => 'headers array is an empty array',
+        ],
       ];
     }
 
-    // Reference the list of expected headers.
+    // Get the list of expected headers.
     $expected_headers = $this->getHeaders();
 
     foreach ($expected_headers as $header) {
-      // Each header name in the expected headers array will be verified for both
-      // index order and presence in the headers provided. Terminate varification
-      // on the first instance of failed result.
-
-      // Take one item from the headers input and compare it to
-      // the current expected header.
+      // Compare expected headers and input headers. Return a failed
+      // validation status on the first instance of a mismatch.
```

Comment:
This comment confused me here. I think it makes more sense to move it above the foreach rather then directly underneath it.

Reply:
Addressed in commit [9fc71a7](https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/119/commits/9fc71a7363fe75be35f7b6bbd44061aaaa8e6f48)!